packages/autocomplete-js/src/__tests__/autocomplete.test.ts

```diff
@@ -156,6 +163,200 @@ describe('autocomplete-js', () => {
+    const renderEmpty = jest.fn(({ root }) => {
+      const div = document.createElement('div');
+      div.innerHTML = 'No results render';
+
+      root.appendChild(div);
+    });
+
+    document.body.appendChild(panelContainer);
+    autocomplete<{ label: string }>({
+      container,
+      panelContainer,
+      getSources() {
+        return [
+          {
+            getItems() {
+              return [];
+            },
+            templates: {
+              item({ item }) {
+                return item.label;
+              },
+            },
+          },
+        ];
+      },
+      renderEmpty,
+    });
+
+    const input = container.querySelector<HTMLInputElement>('.aa-Input');
+
+    fireEvent.input(input, {
+      target: { value: 'aasdjfaisdf' },
+    });
+    input.focus();
+
+    await waitFor(() => {
+      expect(
+        panelContainer.querySelector<HTMLElement>('.aa-Panel')
+      ).toBeInTheDocument();
+    });
+
+    expect(renderEmpty).toHaveBeenCalledWith({
+      root: expect.anything(),
+      state: expect.anything(),
+      sections: expect.anything(),
+    });
+
+    expect(
+      panelContainer.querySelector<HTMLElement>('.aa-Panel')
+    ).toHaveTextContent('No results render');
+  });
+
+  test('renders empty template over renderEmpty method on no results', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+    autocomplete<{ label: string }>({
+      container,
+      panelContainer,
+      getSources() {
+        return [
+          {
+            getItems() {
+              return [];
+            },
+            templates: {
+              item({ item }) {
+                return item.label;
+              },
+              empty() {
+                return 'No results template';
+              },
+            },
+          },
+        ];
+      },
+      renderEmpty({ root }) {
+        const div = document.createElement('div');
+        div.innerHTML = 'No results render';
+
+        root.appendChild(div);
+      },
+    });
+
+    const input = container.querySelector<HTMLInputElement>('.aa-Input');
+
+    fireEvent.input(input, {
+      target: { value: 'aasdjfaisdf' },
+    });
+    input.focus();
+
+    await waitFor(() => {
+      expect(
+        panelContainer.querySelector<HTMLElement>('.aa-Panel')
+      ).toBeInTheDocument();
+    });
+
+    expect(
+      panelContainer.querySelector<HTMLElement>('.aa-Panel')
+    ).toHaveTextContent('No results template');
+  });
+
+  test('overrides the default implementation of shouldPanelShow', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+    autocomplete<{ label: string }>({
+      container,
+      panelContainer,
+      shouldPanelShow: ({ state }) => getItemsCount(state) < 0,
```

Comment:
No need to import `getItemsCount`—wouldn't this be simpler?

```suggestion
      shouldPanelShow: () => false,
```

Reply:
It was mostly to make it more realistic but indeed it is!